lib/player.js

```diff
@@ -5521,14 +5521,19 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
    * @private
    */
   tryToRecoverFromError_(error) {
-    if (error.code !== shaka.util.Error.Code.HTTP_ERROR ||
-      error.category !== shaka.util.Error.Category.NETWORK) {
+    if ((error.code != shaka.util.Error.Code.HTTP_ERROR &&
+      error.code != shaka.util.Error.Code.SEGMENT_MISSING) ||
+      error.category != shaka.util.Error.Category.NETWORK) {
       return false;
     }
 
-    const maxDisabledTime = this.config_.streaming.maxDisabledTime;
+    let maxDisabledTime = this.config_.streaming.maxDisabledTime;
     if (maxDisabledTime == 0) {
-      return false;
+      if (error.code == shaka.util.Error.Code.SEGMENT_MISSING) {
+        maxDisabledTime = 1;
```

Comment:
I was unsure if this was the best way to handle gaps, but [the spec](https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-6.3.3) says:

>    The client SHOULD NOT attempt to load Media Segments that have been
>    marked with an EXT-X-GAP tag, or to load Partial Segments with a
>    GAP=YES attribute.  Instead, clients are encouraged to **look for
>    another Variant Stream of the same Rendition which does not have the
>    same gap**, and play that instead.

So I think this will do.  A comment might be in order, though, to explain that this is what we are doing and why.  Otherwise, someone like me might start to wonder if gaps are expected to appear in all renditions in the same spots, in which case seeking over the gap would have been the more appropriate resolution.

I also think we should handle GAP=YES in partial segments, if we're not already.

Reply:
Done! Added the support and comment